### PR TITLE
feat: add ha_manage_theme tool and themes section in ha_get_system_health (closes #1580)

### LINKED ISSUE #1580: [FEATURE] Theme workflow: read/verify glue around the file-based path (low priority)
### What would you like to see?

Filing this with modest expectations – the file-based theme path works today, and I'd consider "document the recipe" or even a wontfix a reasonable outcome. Writing it down mainly so the gaps are on record. As far as I can tell there has been no user demand for theme features so far (no prior issue or PR asks for theme management).

What already works:

1. Create/edit a theme: `ha_write_file` into `themes/` (an allowed write directory), then `ha_reload_core` with the `themes` target.
2. Apply a theme: the `frontend.set_theme` service via `ha_call_service`.

The gaps are the glue around that recipe:

1. **No way to list installed/active themes.** The frontend exposes them through the `frontend/get_themes` WebSocket command, which no tool surfaces. Without it, "which themes exist, and which one is active?" is guesswork – listing `themes/` shows files, not what is actually loaded or applied.
2. **The `frontend.themes` include prerequisite is invisible.** Files in `themes/` only load if `configuration.yaml` wires the directory up under the `frontend:` key, e.g.:

   ```yaml
   frontend:
     themes: !include_dir_merge_named themes
   ```

   The server can neither check that (`configuration.yaml` is not a readable path) nor – deliberately – create it (`frontend` is not an allowed YAML key, consistent with the maintainer position on `configuration.yaml` edits in #581 / #1034, which this issue does **not** propose to change). When the include is missing, the write+reload recipe silently does nothing, and the assistant cannot diagnose why.

Smallest useful increment, for discussion: surface `frontend/get_themes` read-only wherever it fits best (possibly as another `ha_get_system_health` include section), which would cover both gaps at once – it lists themes *and* its output reveals whether custom themes are wired up at all – plus a short documented recipe for the write+reload+apply flow. Explicitly not proposed: a theme CRUD tool family or any `configuration.yaml` write path.

### Why do you need this?

Theme requests ("make my dashboard dark / high-contrast") are a plausible assistant task, and the building blocks exist – but without the list/verify glue the flow is trial-and-error: write a file, reload, and if nothing changes there is no signal whether the YAML is wrong, the include is missing, or the theme simply was not applied.

### Additional context

**Finding from looking into this:** `themes/` has been a deliberately allowed write directory since the filesystem tools were introduced (#194) and was carried through later scoping changes (#1034). The file-based mechanism is the intended path; this request is only about the read/verify glue around it.

### PR DESCRIPTION
**Pull request by Patch76**
## What does this PR do?

Implements the scope agreed in #1580 for theme support:

- **New `ha_manage_theme` tool** (`tools_themes.py`): `action='list'` returns installed theme names plus the current light/dark defaults via `frontend/get_themes`; `action='set'` selects the backend default theme through `frontend.set_theme` (optional `mode='light'|'dark'`) and re-reads the defaults to verify the result. The docstring routes theme file creation/editing to `ha_config_set_yaml` (custom theme YAML) and HACS (community themes).
- **`themes/*.yaml` scope in `ha_config_set_yaml`** (per review): the `edit_yaml_config` component service (manifest 0.8.0) accepts standalone theme files keyed by theme name (add/replace/remove via the shared single-key apply logic) and triggers `frontend.reload_themes` after each successful write; a failed reload degrades to `post_action: "reload_available"` and surfaces as a top-level warning in the tool response. Documented caveat: themes only load when configuration.yaml carries the `frontend: themes:` include, which the tool cannot add.
- **New `themes` include section in `ha_get_system_health`**: same listing surface (names + defaults, not the full CSS variable dicts) for diagnostics workflows. Listing logic is shared via a `summarize_theme_listing` helper in `util_helpers.py`, guarded against degraded frontend responses (`themes: null` or non-dict shapes).
- **Tests**: unit tests for the new tool, the shared helper, the component themes scope, and the wrapper warning surfacing, plus E2E tests covering list (seeded `Test Theme A`/`Test Theme B`, `count == 2`), set (built-in `default`/`none` paths and real-theme A→B switching with light/dark independence), unknown-theme validation errors, the missing-parameter error path, a `ha_get_system_health(include="themes")` test in the existing system E2E suite, and the create→reload→appear→remove lifecycle through `ha_config_set_yaml`. New `themes` pytest marker.
- Tool docs (README table, `site/src/data/tools.json`, addon DOCS.md) regenerated via `extract_tools.py`.

Closes #1580.

Testing notes: full unit suite and ruff/mypy pass locally; E2E tests run in CI (no Docker available in my local environment).

## Type of change
- [ ] 🐛 Bug fix
- [x] ✨ New feature
- [ ] 📚 Documentation
- [ ] 🔧 Maintenance/refactor
- [ ] 🧪 Tests only
- [ ] 💥 Breaking change

## Testing
- [ ] I have tested these changes with a LLM agent
- [x] All automated tests pass (`uv run pytest`)
- [x] Code follows style guidelines (`uv run ruff check`)

## Checklist
- [x] I have updated documentation if needed

### CODE REVIEW

File: src/ha_mcp/tools/tools_themes.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+            raise
+        except Exception as e:
+            exception_to_structured_error(
+                e,
+                context={"action": action, "theme_name": theme_name, "mode": mode},
+                suggestions=[
+                    "Call ha_manage_theme(action='list') to see installed themes",
+                    "Verify the Home Assistant connection",
+                ],
+            )
```
[the remark this answers is not in the document]
Comment (Patch76) [reply src/ha_mcp/tools/tools_themes.py:185]: ✅ Fixed in d82fd597 — added the explicit `return None` after `exception_to_structured_error` (matching the tools_system.py pattern). This also clears the CodeQL implicit-return finding on the same line.

File: tests/src/e2e/workflows/themes/test_manage_theme.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+instances ship without custom themes, so listing asserts structure rather
+than content, and the set path exercises the built-in 'default' theme plus
+Home Assistant's call-time validation of unknown names.
+"""
+
+import logging
+
+import pytest
+
+from ...utilities.assertions import parse_mcp_result, safe_call_tool
```
[the remark this answers is not in the document]
Comment (Patch76) [reply tests/src/e2e/workflows/themes/test_manage_theme.py]: ✅ Adopted in 6cfb556f — imported top-level alongside the existing assertions imports (multi-line form).

File: tests/src/e2e/workflows/themes/test_manage_theme.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+            {"action": "set", "theme_name": unknown},
+        )
+
+        assert data.get("success") is False, (
+            f"Setting an unknown theme should fail: {data}"
+        )
+        error = data.get("error") or {}
+        assert unknown in str(error.get("message", "")), (
+            f"Error message should name the unknown theme: {data}"
+        )
```
[the remark this answers is not in the document]
Comment (Patch76) [reply tests/src/e2e/workflows/themes/test_manage_theme.py]: ✅ Fixed in 6cfb556f (test_manage_theme.py:97). The missing-param test below intentionally keeps dict access — it asserts the error *code*, which the helper doesn't expose.

File: tests/src/e2e/workflows/system/test_system_tools.py [hunk 4]
```diff
@@ -606,6 +606,45 @@ async def test_get_system_health_unknown_include_warns(self, mcp_client):
...
+            "ha_get_system_health", {"include": "themes"}
+        )
+        data = parse_mcp_result(result)
+
+        if not data.get("success"):
+            error_msg = str(data.get("error", ""))
+            if "not available" in error_msg.lower():
+                pytest.skip("system_health not available in test environment")
+            else:
+                pytest.fail(f"Get system health with themes failed: {error_msg}")
```
[the remark this answers is not in the document]
Comment (Patch76) [reply tests/src/e2e/workflows/system/test_system_tools.py:634]: ✅ Fixed in 6cfb556f (test_system_tools.py:630), with a top-level import instead of the suggested function-local one. The same manual pattern predates this PR at six more sites in this file (e.g. L501/L530/L560); left as-is to keep the diff scoped to the themes feature — happy to sweep them here if preferred.

**Comment by Patch76** [comment]
## Implementation Summary

**Choices Made:**

- **One multi-modal `ha_manage_theme` tool (action=list/set)** instead of separate tools or burying set elsewhere, per the maintainer decision on issue #1580; the read path additionally ships as `include="themes"` on `ha_get_system_health` so health snapshots can carry the theme inventory without an extra call. Both surfaces share one summarizer (`summarize_theme_listing` in `util_helpers.py`, with a None-guard for HA's `themes: null` first-boot state) so their payload shape cannot drift.
- **Set path delegates validation to Home Assistant**: `frontend.set_theme` is called as-is and HA validates the theme name at call time; the tool re-raises HA's rejection with the offending name spelled out; on success it re-reads `frontend/get_themes` so the response reflects the actually-applied state instead of echoing the request.
- **`destructiveHint: true` annotation**: the annotations contract test requires exactly one truthy hint, and setting a frontend-wide default theme mutates user-visible state (precedent: `ha_manage_pipeline`).
- **Tests**: 9 unit tests for the tool + 6 for `_fetch_themes` (WS mock includes `default_dark_theme` unconditionally - HA always returns the key, None when unset), 5 e2e tests under `workflows/themes/` (list shape, set default, dark-mode set with independence check and `none` cleanup, unknown theme, missing param) plus a themes-include e2e on system health; new `themes` marker registered in pytest.ini.
- **Rebased onto master** after #1582/#1584 landed (both touched files this PR also touches); regenerating the tool docs on the rebased tree produced a zero diff, confirming the merge of the generated artifacts (`tools.json` now lists 84 tools).
- **Bot review round adopted**: error-message assertions in the new e2e tests go through the `extract_error_message` helper; the missing-param test keeps dict access deliberately because it asserts the error *code*, which the helper does not expose. Six pre-existing manual-extraction sites in `test_system_tools.py` predate this PR and were left untouched for diff tightness (flagged in the thread, can sweep on request).

**Problems Encountered:**

- **Unknown-theme rejection surfaced as a generic HA error** in e2e (`test_set_unknown_theme_fails`): HA's `service_not_found`-style message did not name the theme. Fixed by catching `HomeAssistantAPIError` on the set path and re-raising a structured error that names the rejected theme.
- **CodeQL flagged mixed implicit/explicit returns** in the new exception handler; resolved with an explicit `return None` after `exception_to_structured_error`.
- **Local environment limits**: three unit-test modules are not loadable in this Alpine venv (starlette test-client needs an httpx the resolver refuses here) and e2e needs Docker, which this box lacks - both validated by CI instead; the remaining local unit suite passes (4456 passed / 77 skipped on the rebased tree).

**Review by kingpanther13 — CHANGES_REQUESTED** [review]
Reviewed against the live connector — list, system-health themes section, set light/dark, `none` reset, unknown-theme rejection, and the validation path all behave correctly, and CI is green. Before merge, the theme-editing story needs to match the direction from #1580, and the real-set path needs e2e coverage. Folding all of this into this PR so #1580 closes complete:

**1. Route theme *editing* through `ha_config_set_yaml` and/or HACS — not the read/write file tools.**
Per the issue discussion, theme creation/editing should go through `ha_config_set_yaml` (for custom theme YAML) and HACS (for community themes, since there's no UI equivalent). The read/write file tools should not be the steered path. Please:
- Revert the `ha_write_file` docstring change (the write→reload→verify theme recipe) — it points at the wrong tool.
- In the `ha_manage_theme` docstring, replace the `ha_write_file` edit pointer with `ha_config_set_yaml` (custom YAML edits) and `ha_manage_hacs` (community install).

**2. Extend `ha_config_set_yaml` to manage theme files.**
Today it edits top-level keys in `configuration.yaml`/`packages/*.yaml`. Themes are a different shape: standalone `themes/<name>.yaml` files merged via `frontend: themes: !include_dir_merge_named themes`, keyed by theme name. Add `themes/*.yaml` as a supported standalone-file scope (add/replace/remove a theme), and trigger `frontend.reload_themes` after the write so the change actually loads (a core reload won't pick it up). This is the editing path the issue settled on; it's beta-gated like the rest of the tool, which is fine.

**3. Add an e2e test that switches between real installed themes.**
The current `set` e2e only exercises built-in `default`/`none`, which hit HA's special `VALUE_NO_THEME`/`DEFAULT_THEME` branch — not the `_update_hass_theme(name, ...)` path a real named theme uses. There's no real-HA proof that setting an installed theme makes `default_theme` become that name, nor an A→B switch. `tests/initial_test_state/configuration.yaml` (lines 6-7) already wires up the `frontend: themes:` include but seeds no `themes/` dir. Seed one or two theme files into `tests/initial_test_state/themes/` (e.g. `Test Theme A`, `Test Theme B`) so they load at boot, then cover: `list` shows both (`count == 2`) → set A → assert `default_theme == "Test Theme A"` → set B → assert it flipped → light/dark independence with a real name → cleanup. If the `config_set_yaml` themes support from item 2 lands, exercise the create→reload→appear path through it here too.

**4. `action="set"` return shape.** `list` returns `{success, data}`; `set` returns `{success, data, message}`, where `message` only restates `data.theme`/`data.mode`. No other tool puts `message` beside a `data` envelope — drop it.

**5. `none` reset phrasing.** `theme_name="none"` returns `"theme":"none"` / `"Theme 'none' applied as ... default"`, but HA *resets* to the built-in default. The structured `default_*_theme: null` is right; reword the prose so it reads as a reset, not an apply.

**6. `summarize_theme_listing` shape guard.** `raw_themes.get("themes") or {}` handles `null`, but a list-shaped `themes` would `AttributeError` on `.keys()`. Add an `isinstance(..., dict)` guard so an upstream shape change degrades to an empty listing.

**7. Two unit assertions.** (a) No unit test asserts a *light*-mode set leaves the *dark* default untouched (only e2e covers independence). (b) The `result["themes"] = {"error": ws_error}` branch when the health WebSocket is unavailable (`tools_system.py:489`) isn't directly asserted. Both one-liners.

**Comment by Patch76** [comment]
All seven items are folded in across b46d72c1 and c2e41297. Item by item:

**1. Theme editing routed through `ha_config_set_yaml` / HACS.** The `ha_write_file` write→reload→verify recipe is reverted, and its pre-existing "Create a theme file" example plus the "Adding theme files" bullet are gone with it (same steering class). The `ha_manage_theme` docstring (and the module docstring) now point custom-theme edits at `ha_config_set_yaml` and community installs at `ha_manage_hacs`.

**2. `ha_config_set_yaml` manages theme files.** `edit_yaml_config` (component 0.8.0) accepts `themes/*.yaml` as a standalone-file scope: `yaml_path` is the theme name (dotted names rejected), `content` is the variable mapping only, and add/replace/remove reuse the existing single-key apply logic (`custom_components/ha_mcp_tools/__init__.py:704`, validation at `:1075`). The component triggers `frontend.reload_themes` after each successful write (`:981`); a failed reload degrades to `post_action: "reload_available"` plus `reload_error`, and the wrapper surfaces that as a top-level warning instead of a bare success (`src/ha_mcp/tools/tools_yaml_config.py:364`). One caveat is documented in the docstring: themes only load when configuration.yaml carries the `frontend: themes:` include, which this tool cannot add (`frontend` is deliberately not an allowed key).

**3. Real-theme e2e.** Seeded `Test Theme A`/`Test Theme B` into `tests/initial_test_state/themes/`. `test_switch_between_real_themes` (themes/test_manage_theme.py:93) covers set A → `default_theme == "Test Theme A"` → flip to B → dark-mode set with a real name leaving the light default untouched, with try/finally cleanup back to `default`/`none`. The list test now asserts both seeded names and `count == 2`. The create→reload→appear→remove lifecycle runs through `ha_config_set_yaml` in `TestYamlConfigThemesIntegration` (filesystem/test_yaml_config.py:1066), asserting `post_action == "reload_performed"` on both add and remove.

**4. `action="set"` return shape.** The `message` field is gone; `set` returns `{success, data}` like `list`.

**5. 'none' reset phrasing.** The "Theme 'none' applied ..." prose lived in that `message` field, so it went with it; the remaining parameter description already reads as a reset.

**6. `summarize_theme_listing` shape guard.** A non-dict `themes` value now degrades to an empty listing (isinstance guard plus unit test).

**7. Two unit assertions.** Added `test_set_light_mode_leaves_dark_default_untouched` and `test_themes_section_reported_unavailable_when_ws_fails` (the `result["themes"] = {"error": ...}` branch in tools_system.py).

**8. Response-contract note (beyond the asked items).** Theme edits through `ha_config_set_yaml` can now carry a top-level `warnings: list[str]`: it appears exactly when the file write succeeded but the `frontend.reload_themes` call failed (the payload then also has `post_action: "reload_available"` plus `reload_error`). A fully successful edit carries no `warnings` key.

Beyond the seven items, the pass also added unit coverage for remove-last-theme (the file degrades to an empty mapping), adding a second theme to an existing file, and the backup=True path, and fixed the e2e created-theme content which had the theme name nested twice. Component manifest bumped to 0.8.0; tool docs regenerated.

**Comment by Patch76** [comment]
CI note: the `HAOS E2E Tests (inaddon)` check is red on this branch, but it's a pre-existing test-infrastructure failure unrelated to this PR's code, and it is not a required check — the three required checks (`E2E Validation (ubuntu-latest)`, `E2E Validation (ubuntu-24.04-arm)`, `Docker & Add-on Validation`) are green.

Root cause: this branch's content changes the HAOS-image cache key, and the resulting cached base image lacks the dev addon's local-store source directory, so the inaddon fixture's `addons/{slug}/update` call returns "No update available" and the suite ERRORs at setup. Filed with full diagnosis and a proposed fix (Supervisor store reload before update) as #1594.

**Review by kingpanther13 — CHANGES_REQUESTED** [review]
Re-reviewed the two update commits against the live connector and the full toolkit. All seven prior items are correctly addressed — `ha_write_file` recipe reverted and re-pointed to `ha_config_set_yaml`+HACS, `config_set_yaml` extended to `themes/*.yaml` with auto-reload, real theme-A→B e2e plus seeded fixtures and a create/remove lifecycle e2e, redundant `message` removed (confirmed live), `isinstance` guard, and both unit assertions. The theme-edit path is solid: path traversal is safe, content validation is sound, and the `frontend.reload_themes` partial-failure handling (distinct `post_action`, `reload_error`, warning log + actionable `warnings[]`) is exactly right.

Two items before merge:

**1. Bump `MIN_COMPONENT_VERSION` to `0.8.0` (`src/ha_mcp/tools/tools_filesystem.py:56`).**
The component manifest bumped `0.7.0`→`0.8.0` for the new `themes/*.yaml` allowlist, but the server-side floor stayed `0.7.0` and isn't touched in this PR. The function's own docstring (lines 132-138) says to bump it in lockstep when the server depends on "a new accepted yaml_path key" — `themes/*.yaml` is exactly that. Without it, a user on the new server with a <0.8.0 component sends `file="themes/x.yaml"`, reaches the old handler, and gets the misleading `"File 'themes/x.yaml' is not allowed. Only configuration.yaml and packages/*.yaml are supported."` instead of the actionable "update via HACS" prompt this gate exists for. The bump is safe to ship because the 0.8.0 component is fully backward-compatible (the new paths only fire on `themes/*.yaml` requests; existing `configuration.yaml`/`packages` calls are unchanged), so requiring it never breaks a user who updates.

**2. Loosen the exact theme `count` assertion (`tests/src/e2e/workflows/themes/test_manage_theme.py:99`).**
`assert listing["count"] == 2` couples this module to the shared per-worker container's global theme state. The `config_set_yaml` lifecycle test adds and removes a theme in a `finally`; if that cleanup ever fails, a later `count == 2` read on the same worker sees 3 and fails far from the real cause. The test already asserts both seeded names explicitly, so `>= 2` keeps the seeded-presence guarantee without the cross-module coupling.

Three low-severity items, take or leave:
- `themes/.yaml` passes the `themes/*.yaml` allowlist, but HA's `!include_dir_merge_named` skips dotfiles — the tool would report `reload_performed` for a theme HA silently drops. Rejecting a normalized basename starting with `.` closes it.
- "reloads themes automatically" (`tools_themes.py` docstring) reads as unconditional; the reload is best-effort and degrades to `reload_available`+`reload_error` on failure. Consider "attempts an automatic reload."
- The `post_action` sentence in the `ha_config_set_yaml` docstring lists "automation, script, scene, and theme files support reload" — for themes the tool *performs* the reload, for the others it only *advertises* the service. Worth making that distinction explicit.

**Comment by Patch76** [comment]
All five items from the second-pass review are addressed in 8b040a24.

## Blocking

**1. MIN_COMPONENT_VERSION bumped to 0.8.0** (`src/ha_mcp/tools/tools_filesystem.py:60`).
The floor now matches the component manifest, so a server on this release talking to
a <0.8.0 component gets the actionable "update via HACS" prompt instead of the old
handler's misleading "not allowed" message on `themes/<name>.yaml`. The gate is
symbol-referenced in the unit tests so they track the bump automatically (the one
hardcoded "too old" fixture uses 0.0.1).

**2. List-count assertion loosened to `>= 2`** (`tests/src/e2e/workflows/themes/test_manage_theme.py:49`).
The two seeded names are still asserted explicitly above it, so fixture presence is
unchanged; `>= 2` drops the coupling to the shared per-worker container's global theme
state (the config_set_yaml lifecycle test adds/removes a theme on the same worker).
Minor: in the reviewed commit the `== 2` assertion was at :44, not :99.

## Low-severity (all three taken)

- **Dotfile theme basenames rejected** (`custom_components/ha_mcp_tools/__init__.py:719`).
  `themes/.yaml` matched the `themes/*.yaml` allowlist (fnmatch's `*` matches a leading
  dot) but HA's `!include_dir_merge_named` skips dotfiles, so the write would have
  reported a phantom `reload_performed`. The handler now rejects a dotfile basename up
  front, with a regression test (`test_theme_dotfile_basename_rejected`).
- **"attempts an automatic theme reload"** (`tools_themes.py`) — reworded from "reloads
  themes automatically", since the reload is best-effort and degrades to
  `reload_available` + `reload_error`.
- **post_action docstring distinction** (`tools_yaml_config.py`) — now states that for
  `themes/*.yaml` the tool performs `frontend.reload_themes` itself (`reload_performed`),
  whereas template/mqtt/group/automation/script/scene only advertise the reload service
  (`reload_available`, caller triggers it).

`tools.json` regenerated for the two docstring changes. Unit suite + ruff + mypy green locally.

**Review by kingpanther13 — CHANGES_REQUESTED** [review]
Re-reviewed commit 8b040a24 with the full toolkit. All five items from the previous pass are correctly and completely addressed — `MIN_COMPONENT_VERSION`→`0.8.0`, the `count == 2`→`>= 2` loosening (presence still guaranteed by the seeded-name asserts), the dotfile guard plus its unit test, and both docstring clarifications (verified accurate against the component and `const.py`). No regressions; the version-gate tests use the symbol + `0.0.1`, so the bump doesn't break them.

Two items before merge:

**1. Broaden the dotfile guard to dot-prefixed directories (`custom_components/ha_mcp_tools/__init__.py`, the new `is_theme and os.path.basename(normalized).startswith(".")` check).**
The guard is basename-only, so it closes `themes/.hidden.yaml` but not a dot-prefixed subdirectory like `themes/.foo/bar.yaml`: that path passes the `themes/*.yaml` allowlist (fnmatch's `*` spans `/`, and nested theme files are legitimately supported), its basename `bar.yaml` isn't a dotfile, so the guard doesn't fire — the file is written and the handler reports `reload_performed`. But HA's loader skips it: `annotatedyaml`'s `_find_files` does `dirs[:] = [d for d in dirs if _is_file_valid(d)]`, pruning dot-prefixed directories from the walk (`_is_file_valid` = `not name.startswith(".")`). So the theme never loads — the same phantom-`reload_performed` silent failure this guard exists to prevent, one directory level up. Mirror HA's logic: reject if *any* path segment under `themes/` starts with `.`, not just the basename. That subsumes the current check.

**2. Align the new test's docstring with its payload (`tests/src/unit/test_yaml_themes.py`, `test_theme_dotfile_basename_rejected`).**
The docstring opens by describing `themes/.yaml` but the call exercises `themes/.hidden.yaml`. Both are valid rejection cases — only the prose names a different example than the code uses. One-word fix.

**Comment by Patch76** [comment]
Both items addressed in 7dfc4c18.

**1. Guard broadened to dot-prefixed directories** (`custom_components/ha_mcp_tools/__init__.py`). Confirmed against the loader: `annotatedyaml._find_files` filters every walked directory and file basename through `_is_file_valid` (`return not name.startswith(".")`), so a dot-prefixed directory like `themes/.foo/bar.yaml` is pruned exactly like a dot-prefixed file. The guard now rejects a theme path if ANY segment starts with `.`, so `themes/.foo/bar.yaml` and `themes/sub/.hidden.yaml` are caught alongside `themes/.hidden.yaml`, while legitimate nested paths (`themes/sub/dark.yaml`) and mid-name dots (`themes/my.theme.yaml`) still pass. New regression tests cover the dot-directory reject and the mid-name-dot allow.

**2. Test docstring aligned** (`tests/src/unit/test_yaml_themes.py`). `test_theme_dotfile_basename_rejected` now names `themes/.hidden.yaml` to match its payload.

**Review by kingpanther13 — APPROVED** [review]
All feedback across the review rounds is addressed and verified: the seven initial items, the `MIN_COMPONENT_VERSION`→`0.8.0` bump and the loosened theme-count assertion, and the theme dotfile guard broadened to dot-prefixed directories (mirroring HA's `_find_files` segment pruning). Theme editing routes through `ha_config_set_yaml` + HACS rather than the read/write file tools; the 0.8.0 component change is additive and backward-compatible. Coverage spans unit and e2e — real-theme A→B switch, the `config_set_yaml` create→reload→remove lifecycle, and dot-segment rejection with a mid-name-dot regression pin. Toolkit re-review came back clean. Approving.
